Source/WebKit/Shared/IPCTesterProxy.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(IPC_TESTING_API)
+
+#include "MessageReceiver.h"
+
+namespace IPC {
+class Connection;
+}
+
+namespace WebKit {
+
+// Proxy interface to test various IPC activities.
+class IPCTesterProxy final : public IPC::MessageReceiver {
+public:
+    IPCTesterProxy() = default;
+    ~IPCTesterProxy() = default;
+
+    // IPC::MessageReceiver overrides.
+    void didReceiveMessage(IPC::Connection&, IPC::Decoder&);
```

Comment:
We should use final keyword instead of adding a comment like this.

Reply:
I 100% sure we should mark overrides as `final` whenever possible (or `override` otherwise).

With regards to the comment though, I think this is pretty common style in WebKit2 to indicate which base class the overrides are for, though we usually use `// IPC::MessageReceiver` instead of `// IPC::MessageReceiver overrides.`.